IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -870,6 +973,121 @@ - (BOOL)clearWithContext:(id<MSIDRequestContext>)context
     return YES;
 }
 
+#pragma mark - Access Control Lists
+
+- (id)createAccess:(MSIDCacheKey *)key context:(id<MSIDRequestContext>)context error:(__unused NSError **)error
+{
+    if (key.isShared)
+    {
+        if (!self.accessForSharedBlob)
+        {
+            if (!self.trustedApplications)
+            {
+                self.trustedApplications = [self createAccessForCurrentApp:context error:error];
+            }
+            
+            self.accessForSharedBlob =  [self accessCreateWithChangeACL:self.trustedApplications
+                                                                context:context
+                                                                  error:error];
+        }
+        
+        return self.accessForSharedBlob;
+    }
+    else
+    {
+        if (!self.accessForNonSharedBlob)
+        {
+            NSArray* trustedApps = [self createAccessForCurrentApp:context error:error];
+            self.accessForNonSharedBlob = [self accessCreateWithChangeACL:trustedApps
+                                                                  context:context
+                                                                    error:error];
+        }
+        
+        return self.accessForNonSharedBlob;
+    }
+}
+
+- (NSArray *)createAccessForCurrentApp:(id<MSIDRequestContext>)context error:(__unused NSError **)error
+{
+    SecTrustedApplicationRef trustedApplication = nil;
+    OSStatus status = SecTrustedApplicationCreateFromPath(nil, &trustedApplication);
+    if (status != errSecSuccess)
+    {
+        [self createError:@"Failed to create trusted application for current application path"
+                   domain:MSIDKeychainErrorDomain errorCode:status error:error context:context];
+        
+        return nil;
+    }
+    
+    NSArray *trustedApplications = @[(__bridge_transfer id)trustedApplication];
```

Comment:
SecTrustedApplicationRef translates to NSArray?

Reply:
Yes it does.